### PR TITLE
feat: Fir 11007 adapt region behaviour

### PR DESCRIPTION
**Pull request by yuryfirebolt**
- make region parameter required, when creating a new database
- remove region parameter, when creating a new engine, the region is taken from the database, it should be attached to

### CODE REVIEW

File: tests/unit/test_engine.py [hunk 1]
```diff
@@ -405,7 +374,7 @@ def test_engine_create_happy_path_optional_parameters(
     engines_mock.create.assert_called_once_with(
         name="engine_name",
         spec="C1",
-        region="us-east-2",
```
Comment (ptiurin) [inline tests/unit/test_engine.py:408]: You can set `configure_resource_manager` to create a db object with the `default_region` you just added and check it here.

File: src/firebolt_cli/database.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def database() -> None:
     default="",
 )
 @json_option
-@option("--region", help="Region for the new database", default="us-east-1", type=str)
+@option("--region", help="Region for the new database", required=True, type=str)
```
[the remark this answers is not in the document]
Comment (yuryfirebolt) [reply src/firebolt_cli/database.py:69]: I would rather not do it, and imho DDL should not assume a default region either. Incorrectly set region might have legal consequences, and it should be sole responsibility of the customer to decide in which region the database should store the data.

File: tests/unit/test_database.py [hunk 3]
```diff
@@ -24,11 +24,7 @@ def test_database_create(
     rm, databases_mock, _, _, _ = configure_resource_manager
 
     result = CliRunner().invoke(
-        create,
```
[the remark this answers is not in the document]
Comment (yuryfirebolt) [reply tests/unit/test_database.py:27]: Done

File: tests/unit/test_engine.py (hunk 1 above)
Comment (yuryfirebolt) [reply tests/unit/test_engine.py:408]: Done

**Review by ptiurin — APPROVED** [review]

**Pull request merged**